helm-charts/splunk-otel-collector/templates/_helpers.tpl

```diff
@@ -149,7 +149,12 @@ Define name for the Splunk Secret
 {{- if .Values.secret.name -}}
 {{- printf "%s" .Values.secret.name -}}
 {{- else -}}
-{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
+{{ $name := (include "splunk-otel-collector.name" .) -}}
+{{- if contains $name .Release.Name -}}
+{{- .Release.Name | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
```

Comment:
```suggestion
{{ default (include "splunk-otel-collector.fullname" .) .Values.secret.name }}
```
I think you can just replace directly with the helper function, and can also remove the if/else before this. `splunk-otel-collector.fullname` already evaluates the same conditions with the addition of also respecting the `fullNameOverride` variable which is missing here.